src/topology/config/mod.rs

```diff
@@ -26,6 +26,8 @@ pub struct Config {
 pub struct GlobalOptions {
     #[serde(default = "default_data_dir")]
     pub data_dir: Option<PathBuf>,
+    #[serde(default)]
+    pub dns: Vec<String>,
```

Comment:
What would a user put here?

Reply:
IPv4, IPv6 with optional port, of DNS servers to be used.

Reply:
The trust-dns-resolver configuration for nameservers expects ipaddr and port. Is there a reason that String is easier here? and if so, should we look at allowing String to be used for configuration in the resolver itself?

Reply:
Since port is optional, nor `SocketAddr`, nor `IpAddr` can be used. So String is used. And port is optional since there is a well defined port for DNS that is 53. Combined with that often DNS servers are mentioned with only ip and not port, having port optional would give better UX.

I wouldn't add String to the configuration, as that definition is really loose to expand user base, but in the code I prefer type strictness.